jdbc/src/main/scala/zio/sql/ExprSyntaxModule.scala

```diff
@@ -0,0 +1,27 @@
+package zio.sql
+
+import zio._
+import zio.stream.ZStream
+import zio.schema.Schema
+
+trait ExprSyntaxModule { self: Jdbc =>
+  implicit final class ReadSyntax[A](self: Read[A]) {
```

Comment:
imo this end up being the best end-user syntax.

Reply:
what about `toZIO` ? 
imho It seems weird from user perspective to have to call both `transact` and `run`.
Ideally if we could omit calling `run` somehow....how ZIO JDBC does that?

But overall this is great simplification, I'll let also @jczuchnowski review

Reply:
I'm not sure.

The semantics are that you will run the individual statements when you use run on them, but only commit them after the transact scope is done.
I think we should big naming that makes this very clear.

Maybe `withTransaction` instead of `transact`?